Matter: update sharing from Google|Apple Home to reflect current software (#31008)
home-assistant/home-assistant.github.io: source/_integrations/matter.markdown

### To add a new device using the Android Companion app

> This guide describes how to add a new device that has never been added to your network before. This will use the Bluetooth connection of your phone to add the device.

Reviewer: is this clear enough ? Or do you want to refer to "just unpacked from the box" ?

Author: Well, the device I used wasn't really new new. It was just no longer paired to any server and freshly reset.

<lite-youtube videoid="Fk0n0r0eKcE" videotitle="Add Matter device via Android app in Home Assistant"></lite-youtube>

## Sharing a device from Apple Home or Google Home

Use one of these methods if your Matter device was added to Apple Home or Google Home and you want to control it from both Apple or Google Home and Home Assistant.

### Prerequisites

- Latest version of the Home Assistant Companion App installed.
- Have the Matter device integrated to Apple Home or Google Home respectively
- In Home Assistant, have the Matter integration installed.
  - Go to {% my integrations title="**Settings** > **Devices & services**" %}.
  - Add the **Matter (BETA)** integration.
  - When prompted to **Select the connection method**:
    - If you run Home Assistant OS in a regular setup: select **Submit**.
      - This will install the official Matter server add-on.

>     - If you are already running the Matter server in another add-on, in or a custom container:

Reviewer: ```suggestion
    - If you are running the Matter server in a custom container (not recommended):
```

2. Scroll all the way down and select **Turn On Pairing Mode**.
3. You are now given a setup code, copy this to the clipboard.
4. In the Home Assistant Companion App, go to {% my integrations title="**Settings** > **Devices & services**" %}.
5. Select **Devices** > **Add device** and on the QR code scanner page, select **More options**.
6. From the list, select your device and paste the code you just received from Apple Home.
7. Once the device is connected with Home Assistant, the device page opens and you can control it from there.

<lite-youtube videoid="nyGyZv90jnQ" videotitle="Share Matter device from Apple Home to Home Assistant"></lite-youtube>

### Share a device from Google Home

To allow Home Assistant to control the Matter device that has previously been added to Google Home, follow these steps:

1. In Google Home, open the **Devices** page and select the cogwheel.
2. On the **Settings** page, select your Matter device.
3. Select **Linked Matter apps and services**.
4. Select **Link apps and services** to link the device to Home Assistant.
5. Choose Home Assistant from the list.
   - You are redirected to the Home Assistant Companion app now.
   - Select **Add device**.

>    - **Troubleshooting**: If Home Assistant fails to add the device, check if you have the Matter integration installed.

Reviewer: ```suggestion
   - **Troubleshooting**: If Home Assistant fails to add the device, check if you have the Matter integration installed and the latest version of the Companion app.
```
